### PR TITLE
Add Transfer product

### PR DESCRIPTION
**Pull request by tylernappy**
- [x] Front-end
- [x] Node.js
- [x] Ruby
- [x] Python
- [x] Go
- [x] Java

# Description
* Backend client libraries updated to work with Transfer
  * Node to `9.2.0`
  * Ruby to `14.3.0`
  * Python to `8.2.1`
  * Java to `9.4.0`
* Front-end (working and complete)
  * Updated Node client library to `9.2.0` to work with Transfer
  * Added Transfer to `frontend/src/Components/ProductTypes/Products.tsx`
  * Added Transfer to `frontend/src/dataUtilities.ts`
* Backend updates (`node/index.js`, `ruby/app.rb`):
  * `TRANSFER_ID` variable to store transfer ID in memory
  * If `transfer` is included product in `.env`, authorize and create transfer after successful token exchange
  * `authorizeAndcreateTransfer()` helper function (excluding Java) called after successful token exchange 

# UI Changes
image

### CODE REVIEW

File: ruby/app.rb [hunk 1]
```diff
@@ -446,3 +472,72 @@ def format_error(err)
...
+# This is a helper function to authorize and create a Transfer after successful
+# exchange of a public_token for an access_token. The TRANSFER_ID is then used
+# to obtain the data about that particular Transfer.
+def authorize_and_create_transfer(access_token, client)
+  begin
+    # We call /accounts/get to obtain first account_id - in production,
+    # account_id's should be persisted in a data store and retrieved
+    # from there.
+    accounts_get_request = Plaid::AccountsGetRequest.new({ access_token: access_token })
+    accounts_get_response = client.accounts_get(accounts_get_request)
```
Comment (tylernappy) [inline ruby/app.rb:485]: If I put a stop point and manually insert a valid `account_id`, the rest of the program works. But as is, I get this strange error message:
```
http://localhost:3000/ -> /api/create_link_token
2021-09-18 17:53:16 - RuntimeError - Invalid ENUM value fraud_risk for class #Products:
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/plaid-14.3.0/lib/plaid/models/products.rb:45:in `build_from_hash'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/plaid-14.3.0/lib/plaid/models/products.rb:37:in `build_from_hash'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/plaid-14.3.0/lib/plaid/models/item.rb:299:in `_deserialize'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/plaid-14.3.0/lib/plaid/models/item.rb:250:in `block (2 levels) in build_from_hash'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/plaid-14.3.0/lib/plaid/models/item.rb:250:in `map'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/plaid-14.3.0/lib/plaid/models/item.rb:250:in `block in build_from_hash'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/plaid-14.3.0/lib/plaid/models/item.rb:243:in `each_pair'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/plaid-14.3.0/lib/plaid/models/item.rb:243:in `build_from_hash'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/plaid-14.3.0/lib/plaid/models/item.rb:235:in `build_from_hash'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/plaid-14.3.0/lib/plaid/models/accounts_get_response.rb:204:in `_deserialize'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/plaid-14.3.0/lib/plaid/models/accounts_get_response.rb:158:in `block in build_from_hash'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/plaid-14.3.0/lib/plaid/models/accounts_get_response.rb:148:in `each_pair'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/plaid-14.3.0/lib/plaid/models/accounts_get_response.rb:148:in `build_from_hash'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/plaid-14.3.0/lib/plaid/models/accounts_get_response.rb:140:in `build_from_hash'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/plaid-14.3.0/lib/plaid/api_client.rb:247:in `convert_to_type'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/plaid-14.3.0/lib/plaid/api_client.rb:208:in `deserialize'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/plaid-14.3.0/lib/plaid/api_client.rb:85:in `call_api'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/plaid-14.3.0/lib/plaid/api/plaid_api.rb:145:in `accounts_get_with_http_info'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/plaid-14.3.0/lib/plaid/api/plaid_api.rb:93:in `accounts_get'
        app.rb:485:in `authorize_and_create_transfer'
        app.rb:66:in `block in <main>'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1636:in `call'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1636:in `block in compile!'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:987:in `block (3 levels) in route!'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1006:in `route_eval'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:987:in `block (2 levels) in route!'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1035:in `block in process_route'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1033:in `catch'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1033:in `process_route'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:985:in `block in route!'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:984:in `each'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:984:in `route!'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1098:in `block in dispatch!'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1072:in `block in invoke'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1072:in `catch'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1072:in `invoke'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1095:in `dispatch!'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:919:in `block in call!'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1072:in `block in invoke'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1072:in `catch'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1072:in `invoke'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:919:in `call!'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:908:in `call'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/rack-protection-2.0.8.1/lib/rack/protection/xss_header.rb:18:in `call'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/rack-protection-2.0.8.1/lib/rack/protection/path_traversal.rb:16:in `call'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/rack-protection-2.0.8.1/lib/rack/protection/base.rb:50:in `call'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/rack-protection-2.0.8.1/lib/rack/protection/base.rb:50:in `call'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/rack-protection-2.0.8.1/lib/rack/protection/frame_options.rb:31:in `call'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/rack-2.2.3/lib/rack/logger.rb:17:in `call'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/rack-2.2.3/lib/rack/common_logger.rb:38:in `call'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:231:in `call'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:224:in `call'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/rack-2.2.3/lib/rack/head.rb:12:in `call'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/rack-2.2.3/lib/rack/method_override.rb:24:in `call'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/sinatra-2.0.8.1/lib/sinatra/show_exceptions.rb:22:in `call'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:194:in `call'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1951:in `call'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1503:in `block in call'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1730:in `synchronize'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1503:in `call'
        /Users/tnappy/.rvm/gems/ruby-2.7.1/gems/rack-2.2.3/lib/rack/handler/webrick.rb:95:in `service'
        /Users/tnappy/.rvm/rubies/ruby-2.7.1/lib/ruby/2.7.0/webrick/httpserver.rb:140:in `service'
        /Users/tnappy/.rvm/rubies/ruby-2.7.1/lib/ruby/2.7.0/webrick/httpserver.rb:96:in `run'
        /Users/tnappy/.rvm/rubies/ruby-2.7.1/lib/ruby/2.7.0/webrick/server.rb:307:in `block in start_thread'
```

File: go/server.go [hunk 2]
```diff
@@ -126,6 +127,11 @@ var itemID string
 
 var paymentID string
 
+// The transfer_id is only relevant for Transfer ACH product.
```
Comment (daniellacoste) [inline go/server.go:130]:
```suggestion
// The transfer_id is only relevant for the Transfer ACH product.
```

File: go/server.go [hunk 3]
```diff
@@ -126,6 +127,11 @@ var itemID string
...
+// We store the transfer_id in memomory - in produciton, store it in a secure
```
Comment (daniellacoste) [inline go/server.go:131]: two typos (make sure to fix these throughout the PR): 

```suggestion
// We store the transfer_id in memory - in production, store it in a secure
```

**Review by daniellacoste — CHANGES_REQUESTED** [review]

**Review by lindawoo-plaid — APPROVED** [review]
LGTM

**Comment by tylernappy** [comment]
@lindawoo-plaid I added your suggestions not through the Github UI, so it dismissed the review.

Can you please re-review?

**Review by lindawoo-plaid — APPROVED** [review]

**Pull request merged**